gst/nnstreamer/nnstreamer_plugin_api_filter.h

```diff
@@ -211,6 +211,13 @@ typedef struct _GstTensorFilterFramework
        * @param[in/out] private_data A subplugin may save its internal private data here. The subplugin is responsible for alloc/free of this pointer. Normally, close() frees private_data and set NULL.
        * @return 0 if ok. < 0 if error.
        */
+
+  int (*checkHWSupport) (accl_hw hw);
+      /**< Optional. Check if the provided hardware accelerator is supported. This check can be static or dynamic based on framework support. Positive response of this check does not guarantee successful running of model with this accelerator. This check can be performed without opening the framework.
+       *
+       * @param[in] hw backend accelerator hardware
+       * @return 0 if supported. -errno if not supported.
+       */
```

Comment:
Question: Is this called even if it's NOT opened?
If it's supposed to be (or if you may modify the flow accordingly) called after it's opened, you can do this with open.

Suggestion: This should not only applied to HW. It should check the general capability and availability of the given condition.

Reply:
As single API provides `int ml_check_nnfw_availability (ml_nnfw_type_e nnfw, ml_nnfw_hw_e hw, bool *available);`, this function SUPPORTS calling this even if the framework is NOT opened.

Updated `open()` to ensure checking that the passed hw is supported.

Reply:
> Suggestion: This should not only applied to HW. It should check the general capability and availability of the given condition.

Are there any other capabilities other than supported `data types`?

Reply:
Then, Let's rename ```checkHWSupport``` --> ```checkAvailability```. In the future, I feel that we may use GstTensorFilterPropertires instead of accl_hw for this callback; we may use accl_hw instead of const char* for accl_str in GstTensorFilterProperties in the future. Then, we may call ```checkAvailabiliby``` for general availability check for a subplugin without actual model files given.

Reply:
Renamed to `checkAvailability`. 
Using GstTensorFilterProperties in this callback is better - will update later.